### ISSUE TITLE
Rancher-Compose: Improve handling of invalid characters in project name

### ISSUE THREAD

**Issue by thaJeztah** [rancher/rancher#1565]
Downloading and extracting rancher-compose and running it from the downloaded location gives a bad experience for a first-time user;

    INFO[0000] Creating environment rancher-compose-v0.2.1
    ERRO[0000] Failed to open project rancher-compose-v0.2.1: Bad response from [http://myhost:8080/v1/environments], statusCode [422]. Status [422 status code 422]. Body: [{"id":"REDACTED","type":"error","links":{},"actions":{},"status":422,"code":"InvalidCharacters","message":null,"detail":null,"fieldName":"name"}] 
    2015/07/17 17:43:01 Failed to read project: Bad response from [http://myhost:8080/v1/environments], statusCode [422]. Status [422 status code 422]. Body: [{"id":"REDACTED","type":"error","links":{},"actions":{},"status":422,"code":"InvalidCharacters","message":null,"detail":null,"fieldName":"name"}]

Which is the result of Rancher-Compose attempting to create a project called `rancher-compose-v0.2.1` (based on the directory name), which contains invalid characters. A nicer error-message would be good here.

In comparison, to prevent similar issues, docker-compose will (currently) strip-out non `[a-z0-9_]` characters, and create a project called `ranchercomposev021`.

**Comment by ibuildthecloud** [comment]
@thaJeztah Thanks for the pointer.  We will follow what Docker Compose is doing.

**Comment by thaJeztah** [comment]
@ibuildthecloud not sure if my regexp is a 100% correct, but it should be easy to find in the source.

A human-readable error-message would be a good start though ;-)

**Comment by ibuildthecloud** [comment]
We currently support humans that read JSON :)

**Comment by rebyn** [comment]
I came across a weird behavior today and at first read it seems like I should file it here.

I put `rancher-compose` in a folder named `MyApp` and the command below tried to create a new stack (instead of recognizing my existing stack):

```
➜  MyApp  ./rancher-compose upgrade AppV1 AppV2
INFO[0001] Creating environment myapp
```

Then I tried setting the project name with the `-p` tag:

```
➜  MyApp  ./rancher-compose -p="MyApp" upgrade AppV1 AppV2
INFO[0001] Creating environment myapp
```

Not until I renamed my stack to `myapp` (from `MyApp` on the Rancher dashboard) did these commands go through.

**Comment by deniseschannon** [comment]
@rebyn I've created a new issue based on your comment. The issues is how we handle case sensitivity of stack names.

**Comment by deniseschannon** [comment]
Verified in v0.3.0-rc1
